server/ai_process.go

```diff
@@ -1048,6 +1048,7 @@ func submitLiveVideoToVideo(ctx context.Context, params aiRequestParams, sess *A
 		if err != nil {
 			return nil, fmt.Errorf("invalid control URL: %w", err)
 		}
+		// TODO any errors from these funcs should we kill the input stream?
```

Comment:
@j0sh wdyt, I'm thinking we should pass errors back from these publish and subscribe functions below and generally we should kill the input stream (essentially following what you described [here](https://www.notion.so/livepeer/Live-AI-Pipeline-Integration-Points-1510a348568780379b40f42d7f2f2332?pvs=4#1520a34856878041a3abf4e5964d8790))

Reply:
Aren't these functions async? It might take some effort to refactor these, but whatever way is most straightforward sounds good to me. I was even thinking of having it kill the ingest on MediaMTX and letting things propagate down but that is probably a little (too?) indirect. We might just need to have a general cleanup routine that can be called wherever things fail.